### PR TITLE
fix for cancell error supression in multiplexer._connect_all

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

fix issue with termination on connect_all

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/multiplexer.py [hunk 1]
```diff
@@ -274,7 +277,7 @@ async def disconnect(self) -> None:
         self.logger.debug("Multiplexer disconnecting...")
         async with self._lock:
             if self.connection_status.is_disconnected:
-                self.logger.debug("Multiplexer already disconnected.")
+                self.logger.debug("Multiplexer disconnected.")
```
Comment (DavidMinarsch) [inline aea/multiplexer.py:280]: Seems like cheating...double disconnections will not be as easy to notice...

File: aea/multiplexer.py [hunk 2]
```diff
@@ -341,14 +344,13 @@ async def _connect_all(self) -> None:
...
+                if not isinstance(e, (asyncio.CancelledError, CancelledError)):
+                    self.logger.exception(
+                        "Error while connecting {}: {}".format(
+                            str(type(connection)), repr(e)
+                        )
                     )
-                )
-                for c in connected:
-                    await self._disconnect_one(c)
-                break
```
Comment (DavidMinarsch) [inline aea/multiplexer.py:351]: Where is the disconnection now taking place?

Comment (solarw) [reply aea/multiplexer.py:351]: in multiplexer.connect

File: aea/multiplexer.py (hunk 1 above)
Comment (solarw) [reply aea/multiplexer.py:280]: fixed

**Review by DavidMinarsch — APPROVED** [review]
LGTM

File: aea/multiplexer.py [hunk 3]
```diff
@@ -341,14 +344,13 @@ async def _connect_all(self) -> None:
...
                 connected.append(connection_id)
             except Exception as e:  # pylint: disable=broad-except
-                self.logger.error(
-                    "Error while connecting {}: {}".format(
-                        str(type(connection)), str(e)
+                if not isinstance(e, (asyncio.CancelledError, CancelledError)):
+                    self.logger.exception(
+                        "Error while connecting {}: {}".format(
+                            str(type(connection)), repr(e)
+                        )
```
Comment (marcofavorito) [inline aea/multiplexer.py:351]: you could use f-strings

File: aea/multiplexer.py (hunk 2 above)
Comment (marcofavorito) [reply aea/multiplexer.py:351]: makes more sense, thanks

**Review by marcofavorito — APPROVED** [review]
LGTM

**Pull request merged**